### PR TITLE
Added tests, minor fixups

### PR DESCRIPTION
**Pull request by blarghmatey**
Fixed some typos and an edge case on the DH file that was caused by how
the default attributes were defined in the openvpn cookbook. Added
serverspec tests to validate the state of the converged node.

### CODE REVIEW

**Comment by blarghmatey** [comment]
@mjuszczak Care to check this out?

File: attributes/default.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 default['diavpn']['users'] = nil
 
-default['openvpn']['key']['size'] = '2048'
+default['openvpn']['key']['size'] = 2048
```
Comment (mjuszczak) [inline attributes/default.rb]: Oh yeah, that can bite you. Good find.

File: attributes/default.rb [hunk 2]
```diff
@@ -14,9 +14,9 @@
 default['openvpn']['config']['topology'] = 'subnet'
 default['openvpn']['config']['ifconfig-pool-persist'] = '/etc/openvpn/ipp.txt'
 default['openvpn']['config']['tls-auth'] = '/etc/openvpn/keys/ta.key 0'
-default['openvpn']['config']['cipher'] = 'AED-256-CBC'
+default['openvpn']['config']['cipher'] = 'AES-256-CBC'
```
Comment (mjuszczak) [inline attributes/default.rb]: Oops! I've been doing defibrillators recently at the gym. haha.

File: attributes/default.rb [hunk 3]
```diff
@@ -14,9 +14,9 @@
...
 default['openvpn']['config']['client-cert-not-required'] = ''
 default['openvpn']['config']['username-as-common-name'] = ''
-default['openvpn']['config']['plugin'] = 'usr/lib64/openvpn/plugin/lib/openvpn-auth-pam.so openvpn'
-
-default['openvpn']['push_routes'] = %w(172.31.0.0 255.255.0.0)
+default['openvpn']['config']['plugin'] = '/usr/lib64/openvpn/plugin/lib/openvpn-auth-pam.so openvpn'
+default['openvpn']['config']['dh'] = "#{node['openvpn']['key_dir']}/dh#{node['openvpn']['key']['size']}.pem"
+default['openvpn']['push_routes'] = ['172.31.0.0 255.255.0.0',]
```
Comment (mjuszczak) [inline attributes/default.rb]: What's with the comma?

File: test/integration/default/serverspec/default_spec.rb [hunk 4]
```diff
@@ -1,9 +1,44 @@
...
+  it { should be_running }
+  it { should be_enabled }
+end
+
+describe port(1194) do
+  it { should be_listening.with('udp') }
+end
+
+describe iptables do
+  it { should have_rule('-j MASQUERADE').with_table('nat') }
```
Comment (mjuszczak) [inline test/integration/default/serverspec/default_spec.rb]: Nice!!!!

**Comment by mjuszczak** [comment]
:shipit:
